### PR TITLE
ensure classList is defined before calling .contains()

### PR DESCRIPTION
**Pull request by vardani**
## What

Bug

## Why

Change is being made after observing the following error:

```
{snip} ,hasIgnoreClass=function(e,r){for(var t=e.target;t;){var n;if(null===(n=t)||void 0===n?void 0:n.classList.contains(r))return!0;t=t.parentEle {snip}
./node_modules/react-cool-onclickoutside/dist/index.esm.js in e at line 1:3046
./node_modules/@sentry/browser/esm/helpers.js in r at line 60:1
```

## How

Change was implemented by adding an additional check against null/undefined values before calling `.contains()`.

## Checklist

- [x] Documentation added
- [x] Tests
- [x] Typescript definitions updated
- [x] Ready to be merged

### CODE REVIEW

**Comment by wellyshen** [comment]
@vardani Thank you for this PR. Can we simplify the code by using the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to handle it?

File: src/index.ts [hunk 1]
```diff
@@ -24,7 +24,9 @@ const hasIgnoreClass = (e: any, ignoreClass: string): boolean => {
   let el = e.target;
 
   while (el) {
-    if (el?.classList.contains(ignoreClass)) return true;
+    // el.classList != null checks both for null and
+    // undefined since it's not a strict comparison
+    if (el && el.classList != null && el.classList.contains(ignoreClass)) return true;
```
Comment (wellyshen) [inline src/index.ts:29]: @vardani Thank you for this PR. Can we simplify the code by using the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to handle it?

Comment (vardani) [reply src/index.ts:29]: Sure thing, I'll go ahead and update it. Thanks!

**Comment by wellyshen** [comment]
@vardani Looks great 👍🏻

**Comment by wellyshen** [comment]
@all-contributors please add @vardani for bug.

**Pull request merged**